uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -1103,6 +1108,9 @@ define([
           .then(function (entitlements) {
             // create a list of absence types with a `balance` property
             self.absenceTypes = mapAbsenceTypesWithBalance(absenceTypesAndIds.types, entitlements);
+            if (!self.absenceTypes.length) {
+              return $q.reject(NO_ENTITLEMENT_ERROR);
```

Comment:
It is not mandatory, as in if I dont mention anything it will resolve the promise by default. That is how it was working before, I just added condition to reject the promise in certain condition.